.github/workflows/main.yml

```diff
@@ -8,34 +8,42 @@ on:
       - "[0-9]*"
       - "v[0-9]*"
     tags:
-      - v*      
+      - v*
+  workflow_dispatch:
 
 jobs:
   build:
     runs-on: ubuntu-24.04
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.PUBLIC_PUSH_ECR_AWS_KEY }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.PUBLIC_PUSH_ECR_AWS_SECRET }}
     steps:
     - name: Checkout
       uses: actions/checkout@v5
-    - name: Extract repo name
-      shell: bash
-      run: echo "##[set-output name=repo_name;]$(echo ${GITHUB_REPOSITORY} | sed "s/.*\///g" | sed "s/-docker$//")"
-      id: extract_repo_name
+
     - name: Docker meta
       id: docker_meta
       uses: crazy-max/ghaction-docker-meta@v5
       with:
-        images: public.ecr.aws/citizensadvice/${{ steps.extract_repo_name.outputs.repo_name }}
-        tag-sha: true
-    - name: Login to AWS Elastic Container Registry
-      uses: docker/login-action@v3
+        images: public.ecr.aws/citizensadvice/clamav
+        tags: |
+          type=sha
+          type=ref,event=branch
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v5
+      with:
+        aws-region: us-east-1 # ECR Public can only be logged into from the us-east-1 region.
+
+    - name: ECR login
+      id: ecr-login
+      uses: aws-actions/amazon-ecr-login@v2
       with:
-        registry: public.ecr.aws
-        username: ${{ secrets.PUBLIC_PUSH_ECR_AWS_KEY }}
```

Comment:
why moving these as envvars ?

Reply:
These aren't envvars, they are inputs. The `configure-aws-credentials` action doesn't take keys as inputs, it looks for environment variables. I tried setting the env var on the step itself but the job fails: https://github.com/citizensadvice/clamav-docker/actions/runs/18033336165/job/51314567107

Reply:
I mean you moved the inputs to be envvars. But ok.